test/system/user_test.rb

```diff
@@ -0,0 +1,103 @@
+require "application_system_test_case"
+
+class UsersTest < ApplicationSystemTestCase
+  test "Invite, register, confirm, promote, demote, delete" do
+    admin = users(:admin_user)
+    viewer_email = 'viewer@example.com'
+
+    Capybara.using_session(:admin) do
```

Comment:
Would you mind adding a comment at the start of each of these blocks describing what it’s doing conceptually? This is a long function that flips between sessions, and could use description to make it easier to follow.

Reply:
Great idea! I added comments.